pkg/client/kubernetes/client.go

```diff
@@ -359,3 +372,34 @@ func defaultContentTypeProtobuf(c *rest.Config) *rest.Config {
 	}
 	return &config
 }
+
+var _ client.Client = &fallbackClient{}
+
+type fallbackClient struct {
```

Comment:
Basically part 2 of the question in `cache.go`:
For which "cache errors" do we need special handling / fallback logic?

Reply:
Fallbacks are necessary so that you can use a cached client with a [Multinamespace Cache](https://github.com/kubernetes-sigs/controller-runtime/blob/4462fd4ebb88171af826622c1d6b73383aaf2cdf/pkg/cache/multi_namespace_cache.go#L65) but only for namespaces that are allowed for Gardenlet. However, there are other namespaces in the Garden cluster e.g., `garden`, which the Gardenlet is permitted to only GET specific resources, but is never allowed to LIST/WATCH the entire namespace. Hence, we need to prevent that the cached Runtime Client automatically starts a watch for these "restricted" namespaces.

Reply:
Yep, makes sense.
Now I fully understood the rationale :)